docs/config.md

```diff
@@ -0,0 +1,68 @@
+# Terramate Configuration
+
+There is a series of different configurations that can be done
+on Terramate, ranging from avoiding duplication to controlling
+order of execution of stacks.
+
+In order to do so, Terramate works with configuration files named
+**terramate.tm.hcl**. These files can be found in any directory
+of a Terramate project, but some configurations will be defined
+only once for an entire project, while others provide merge/overriding
+strategies, while others control only behavior specific for
+stacks.
+
+Here is defined the different kinds of configurations and
+how you can learn more about them. But before getting into the different
+kinds of configurations, what would be a Terramate project?
+
+A Terramate project is essentially any collection of Terraform code
+organized into stacks. You can have all Terraform code together in a single
+directory but that would defeat the purpose of most Terramate features, so
+usually, you have a set of stacks and maybe a set of Terraform modules if they
+are not maintained in a different repository.
+
+It is not a hard requirement for Terramate to work that the project uses git 
+for version control, but features like change detection do depend on git to
+work and will fail if this requirement is not met.
+
+In general, a Terramate project looks like this:
+
+* A git project
+* The git top-level dir is the project base dir
+* Stacks will be organized as different directories
+* Configuration may be present on any dir
+
+# Project Configuration
+
+Per project configuration can be defined only once at the project base dir.
+
+Available project-wide configurations can be found [here](project-config.md)
+
+# Stack Configuration
+
+Before talking about stack specific configuration, lets define what is a
+Terramate stack:
+
+* A dir that has no child dirs (a leaf in the file system tree)
```

Comment:
i would rather say leaf-config/lead-stack as being the last config inside a directory, there can be other directories inside a leaf-stack  e.g. source code of a cloud/lambda function or examples or documentation.. all that is not another stack.

Reply:
Yeah I meant this here: https://mineirosio.slack.com/archives/C02GFAAAKPT/p1639505791023500?thread_ts=1639504351.015800&cid=C02GFAAAKPT

But then I didn't get the concept of "non-leaf config" from @i4ki question/your answer after my proposal, he asked [here](https://mineirosio.slack.com/archives/C02GFAAAKPT/p1639505942025000?thread_ts=1639504351.015800&cid=C02GFAAAKPT):

"but should we fail if someone puts a stack block in a non-leaf dir?"

And you [answered](https://mineirosio.slack.com/archives/C02GFAAAKPT/p1639505973025500?thread_ts=1639504351.015800&cid=C02GFAAAKPT):

"yes.... stack in non-leaf should be an error..."

Since he was talking about "non-leaf dir" it seemed we actually wanted stacks to be on the fs leaf.